From ed8b05b5d062307fd93e4011c5dfd20f72a5b189 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@polymtl.ca>
Date: Thu, 2 Oct 2008 01:43:57 -0400
Subject: [PATCH] traceprobes

Traceprobes

Menu option to activate tracing probes.

Masami Hiramatsu <mhiramat@redhat.com :

Tested on x86-64.

Changelog :
- Default to m instead of y.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@polymtl.ca>
Acked-by: Masami Hiramatsu <mhiramat@redhat.com>
CC: Alexander Viro <viro@zeniv.linux.org.uk>
CC: 'Peter Zijlstra' <peterz@infradead.org>
CC: "Frank Ch. Eigler" <fche@redhat.com>
CC: 'Ingo Molnar' <mingo@elte.hu>
CC: 'Hideo AOKI' <haoki@redhat.com>
CC: Takashi Nishiie <t-nishiie@np.css.fujitsu.com>
CC: 'Steven Rostedt' <rostedt@goodmis.org>
---
 ltt/Kconfig         |    9 +++++++++
 ltt/Makefile        |    1 +
 ltt/probes/Makefile |    1 +
 3 files changed, 11 insertions(+), 0 deletions(-)
 create mode 100644 ltt/probes/Makefile

diff --git a/ltt/Kconfig b/ltt/Kconfig
index c8e9e72..ac2f4a3 100644
--- a/ltt/Kconfig
+++ b/ltt/Kconfig
@@ -1,5 +1,14 @@
 menu "Linux Trace Toolkit"
 
+config LTT_TRACEPROBES
+	tristate "Compile lttng tracing probes"
+	depends on MARKERS
+	default m
+	help
+	  Compile lttng tracing probes, which connect to the tracepoints when
+	  loaded and format the information collected by the tracepoints with
+	  the Markers.
+
 config LTT_TIMESTAMP
 	bool "LTTng fine-grained timestamping"
 	default y
diff --git a/ltt/Makefile b/ltt/Makefile
index bfe2d09..f6fb6b1 100644
--- a/ltt/Makefile
+++ b/ltt/Makefile
@@ -11,3 +11,4 @@ obj-$(CONFIG_LTT_SERIALIZE)		+= ltt-serialize.o
 obj-$(CONFIG_LTT_STATEDUMP)		+= ltt-statedump.o
 obj-$(CONFIG_HAVE_LTT_UNSTABLE_TSC)	+= ltt-test-tsc.o
 obj-$(CONFIG_LTT_MARKER_CONTROL)	+= ltt-marker-control.o
+obj-$(CONFIG_LTT_TRACEPROBES)		+= probes/
diff --git a/ltt/probes/Makefile b/ltt/probes/Makefile
new file mode 100644
index 0000000..a8b481e
--- /dev/null
+++ b/ltt/probes/Makefile
@@ -0,0 +1 @@
+# LTTng tracing probes
-- 
1.5.5.1

